tensorflow/core/kernels/data/optimize_dataset_op_test.cc

```diff
@@ -45,66 +45,221 @@ class OptimizeDatasetOpTest : public DatasetOpsTestBase {
     TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
     return Status::OK();
   }
+
+  // Create the input dataset as a variant tensor for OptimizeDataset.
+  Status MakeInputDataset(const string& dataset_type, int64 count,
+                          const DataTypeVector& output_types,
+                          const std::vector<PartialTensorShape>& output_shapes,
+                          Tensor* input_dataset) {
+    GraphConstructorOptions graph_opts;
+    graph_opts.allow_internal_ops = true;
+    graph_opts.expect_device_spec = false;
+
+    Tensor range_dataset_tensor;
+    Tensor start = CreateTensor<int64>(TensorShape({}), {-3});
+    Tensor stop = CreateTensor<int64>(TensorShape({}), {3});
+    Tensor step = CreateTensor<int64>(TensorShape({}), {1});
+    TF_RETURN_IF_ERROR(
+        RunFunction(test::function::MakeRangeDataset(),
+                    /*attrs*/
+                    {{"output_types", DataTypeVector({DT_INT64})},
+                     {"output_shapes",
+                      std::vector<PartialTensorShape>{PartialTensorShape({})}}},
+                    /*inputs*/ {start, stop, step}, graph_opts,
+                    /*rets*/ {&range_dataset_tensor}));
+
+    if (dataset_type == "TakeDataset") {
```

Comment:
instead of this ugly if-else, why not have separate method for creating each and have the test case use a lambda for the dataset factory (as opposed to a dataset "type" as a string)

Reply:
Thanks for your comments! Will avoid this kind of if-else in the future.